packages/embeds/embed-core/playwright/tests/action-based.e2e.ts

```diff
@@ -101,9 +98,63 @@ test.describe("Popup Tests", () => {
     });
   });
 
-  todo("Floating Button Test with Dark Theme");
+  test("should open dark embed iframe on floating button click", async ({
```

Comment:
We should do a single test that tests end to end booking.
For testing themes, we should use a different strategy that can be as simple as verifying that the embed has appropriate class on html tag. Refer to https://github.com/calcom/cal.com/pull/9455/files.
This would make the tests faster than the current approach of doing booking e2e.

Reply:
The 2 tests added test e2e booking using floating buttons. When starting the development, I noticed there were 2 TODO tests to test floating buttons with dark/light theme (I thought these were the once that had to be added).

As I understand it then, only 1 e2e test is needed for the floating button, and then 2 more tests that only test the theme of the popup, is that right? (In that case, if I remember correctly, the race condition wouldn't be a problem so the Nth code could be deleted)

Reply:
Yeah exactly.